code/modules/clothing/chameleon.dm

```diff
@@ -69,18 +69,28 @@
 /datum/action/item_action/chameleon/change
 	name = "Chameleon Change"
 	var/list/chameleon_blacklist = list()
+	var/list/chameleon_blacklist_with_subtypes = list()
 	var/list/chameleon_list = list()
 	var/chameleon_type = null
 	var/chameleon_name = "Item"
 
 	var/emp_timer
 
+/datum/action/item_action/chameleon/change/New()
+	..()
+	chameleon_blacklist = typecacheof(chameleon_blacklist)
+	chameleon_blacklist_with_subtypes = typecacheof(chameleon_blacklist_with_subtypes)
+	chameleon_list = typecacheof(chameleon_list)
```

Comment:
Also is it really needed having two vars for blacklisting? It's hard to think a situation where it's needed.